src/voter/mod.rs

```diff
@@ -61,9 +61,9 @@ pub trait Environment<H: Eq, N: BlockNumberOps>: Chain<H, N> {
 	type Timer: Future<Output = Result<(), Self::Error>> + Unpin;
 	/// Associated future type for the environment used when asynchronously computing the
 	/// best chain to vote on. See also [`Self::best_chain_containing`].
-	type BestChain: Future<Output = Result<Option<(H, N)>, Self::Error>> + Send + Sync + Unpin;
+	type BestChain: Future<Output = Result<Option<(H, N)>, Self::Error>> + Send + Unpin;
 	/// The associated Id for the Environment.
-	type Id: Ord + Clone + Eq + ::std::fmt::Debug;
+	type Id: Clone + Eq + Ord + Hash + std::fmt::Debug;
```

Comment:
`Hash` bound addition here requires a major version bump

Reply:
I reverted this change as there's no need to add it here (kept it in the `voter_state` method).